File: frontend/src/app/dto/payment/PaymentResponse.ts
export interface PaymentResponse {
  transactionId: string;    // Mã giao dịch từ backend trả về
  orderId: number;
  userId: number;
  amount: number;
  method: string;
  status: PaymentStatus;    // Enum trạng thái
  message: string;          // Thông báo kết quả (VD: "Payment successful")
  timestamp: string;        // ISO datetime (VD: "2025-10-04T12:00:00Z")
}

export enum PaymentStatus {
  SUCCESS = 'SUCCESS',
  FAILED = 'FAILED',
  PENDING = 'PENDING'
}

File: payment-service/src/main/java/com/vanhuy/payment_service/dto/PaymentResponse.java
package com.vanhuy.payment_service.dto;

import java.time.LocalDateTime;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * DTO trả về cho frontend sau khi xử lý thanh toán.
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
public class PaymentResponse {
    private Long paymentId;           // ID giao dịch thanh toán
    private Integer orderId;             // ID đơn hàng liên quan
    private Double amount;        // Số tiền đã thanh toán
    private String method;            // Phương thức thanh toán
    private String status;            // "SUCCESS" | "FAILED" | "PENDING"
    private String transactionCode;   // Mã giao dịch (nếu có)
    private LocalDateTime timestamp;  // Thời điểm thanh toán
    private String message;           // Thông báo kết quả
}
